### DISCUSSION TITLE
Automatically setup teams and their portfolios

### DISCUSSION THREAD

**Discussion by malice00** [DependencyTrack/dependency-track#2301] [Q&A]
We have been evaluating DependencyTrack for a while and would really like to start using it, but we would like to have some automation on certain parts of the process.

The idea is to have a single managed user that is allowed to upload BOMs, which would be configured in Jenkins. Then all our teams can use these credentials to upload. However, since we have strict policies that teams are not supposed to see other teams' projects, we would like to setup teams in DT as well.
Now, since we are using LDAP for our users and their roles, we were thinking about somehow having a team created (if it doesn't exist yet), map it to the LDAP role and add projects to the portfolio for this team whenever an upload is triggered.

Is it possible to do something like this in DT, or do we need to setup these things manually? How would one go about this?
I've been looking through the REST-API, but seeing most calls need some UUIDs, which our teams would have to figure out again, I'm not quite convinced this is the best way to go... Or is it?

We had an idea of our own to maybe use the property field for this, but that would probably mean something has to be implemented to do something with this property... If nothing like this exists yet, we can maybe look into implementing something like this and create some PRs to get it into an upcoming release.

**Comment by syalioune** [comment]
> Is it possible to do something like this in DT, or do we need to setup these things manually? How would one go about this?

To my knowledge, it is not possible to autocreate teams based on encountered LDAP/OIDC groups. The DT teams should exist so that synchronization could be done with external IdP.
I'm in a kind of similar situation with a closed set of projects, what we do whenever projects should be added to the portfolio :
- Project creation (name, version, tags, custom properties,...)
- Team creation
- API Key creation with the appropriate sets of permissions
- OIDC connect group creation and mapping to the team
- ACL to bind the team to the proper project(s)
- API Key sharing to the project team

We developed an internal CLI tool to perform all those previous steps leveraging the REST API.

> The idea is to have a single managed user that is allowed to upload BOMs, which would be configured in Jenkins. Then all our teams can use these credentials to upload

The issue is see with this is that Team/Project A can upload a SBOM to Project B since they all use the same user who will have access to all DT projects.
